src/Core/Models/Business/TaxInfo.cs

```diff
@@ -37,12 +37,24 @@ public string TaxIdType
 
             switch (BillingAddressCountry)
```

Comment:
It's also important to handle potential issues with input data. If BillingAddressCountry could be in lowercase, ensure it's transformed to uppercase before the lookup.